cc-snapshot

```diff
@@ -246,10 +246,13 @@ else
   OS_AUTH_URL=$(extract_json_key "auth_url_v3" "$JSON_VENDOR_DATA")
 fi
 
-if dmesg | grep -q Hypervisor; then
-  # The current instance is apparently not a baremetal node.
-  >&2 echo "Error: cc-snapshot is only supported for baremetal instances."
-  exit 1
+if  [[ "$DRY_RUN" == true ]]; then 
+  echo "[DRY_RUN] dmesg | grep -q Hypervisor"
+else 
+  if dmesg | grep -q Hypervisor; then
+    echo "Error: cc-snapshot is only supported for baremetal instances." >&2
+    exit 1
+  fi
```

Comment:
We should leave this whole block as-is and exit regardless whether this is dry run or not if this is being run on the wrong instance type. This requiring root (or not, only for testing) is separate from the behavior of exiting (or not) with dry run. We should either refactor this so this command/step is not needed to be run for any testing, or probably simpler, we can just skip this step if `TESTING_SKIP_ROOT_CHECK` is set (we could print a message that we aren't checking for baremetal because that requires sudo).